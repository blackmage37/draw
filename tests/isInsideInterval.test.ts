import isInsideInterval from '../src/utils/isInsideInterval'

describe('isInsideInterval', () => {
  describe.each([
    [0, 1],
    [1, 1],
    [0, Number.POSITIVE_INFINITY],
    [1, 0],
  ])('range: [%s, %s]', (min, max) => {
    const func = isInsideInterval(min, max)
    it.each([0, -0, 0.5, 1, -1, Number.POSITIVE_INFINITY])('value: %s', n => {
      const result = func(n)
      expect(result).toMatchSnapshot()
    })
  })
})
