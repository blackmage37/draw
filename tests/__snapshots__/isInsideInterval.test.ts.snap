// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`isInsideInterval > range: [0, 1] > value: -0 1`] = `true`;

exports[`isInsideInterval > range: [0, 1] > value: -1 1`] = `false`;

exports[`isInsideInterval > range: [0, 1] > value: 0 1`] = `true`;

exports[`isInsideInterval > range: [0, 1] > value: 0.5 1`] = `true`;

exports[`isInsideInterval > range: [0, 1] > value: 1 1`] = `true`;

exports[`isInsideInterval > range: [0, 1] > value: Infinity 1`] = `false`;

exports[`isInsideInterval > range: [0, Infinity] > value: -0 1`] = `true`;

exports[`isInsideInterval > range: [0, Infinity] > value: -1 1`] = `false`;

exports[`isInsideInterval > range: [0, Infinity] > value: 0 1`] = `true`;

exports[`isInsideInterval > range: [0, Infinity] > value: 0.5 1`] = `true`;

exports[`isInsideInterval > range: [0, Infinity] > value: 1 1`] = `true`;

exports[`isInsideInterval > range: [0, Infinity] > value: Infinity 1`] = `true`;

exports[`isInsideInterval > range: [1, 0] > value: -0 1`] = `false`;

exports[`isInsideInterval > range: [1, 0] > value: -1 1`] = `false`;

exports[`isInsideInterval > range: [1, 0] > value: 0 1`] = `false`;

exports[`isInsideInterval > range: [1, 0] > value: 0.5 1`] = `false`;

exports[`isInsideInterval > range: [1, 0] > value: 1 1`] = `false`;

exports[`isInsideInterval > range: [1, 0] > value: Infinity 1`] = `false`;

exports[`isInsideInterval > range: [1, 1] > value: -0 1`] = `false`;

exports[`isInsideInterval > range: [1, 1] > value: -1 1`] = `false`;

exports[`isInsideInterval > range: [1, 1] > value: 0 1`] = `false`;

exports[`isInsideInterval > range: [1, 1] > value: 0.5 1`] = `false`;

exports[`isInsideInterval > range: [1, 1] > value: 1 1`] = `true`;

exports[`isInsideInterval > range: [1, 1] > value: Infinity 1`] = `false`;
